### PR TITLE
PR Updates

### PR DESCRIPTION
**Pull request by jaypoulz**
Modified defaults for lib repo parameters, added test repo overrides reindented, and added archiving for script based tests

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,91 +1,117 @@
...
-              checkout scm
-          }
+    /*********************************************************/
+    /* TEST BODY                                             */
+    /* @param slave               Provisioned slave details. */
+    /*********************************************************/
+    dir('test') {
+      stage ('Download Test Files') {
+        if (params.TEST_REPO) {
+          git url: params.TEST_REPO, branch: params.TEST_VERSION, changelog: false
```
Comment (scoheb) [inline Jenkinsfile]: does this work if TEST_VERSION = '' as specified in defaults?

File: Jenkinsfile [hunk 2]
```diff
@@ -1,91 +1,117 @@
...
-              }
-          }
+      // TODO insert test body here
+      stage ('Run Test') {
+        if (config.runOnSlave) {
+          sh "ansible-playbook ${TEST_DIR}/ansible-playbooks/*/playbook.yml"
+          // TODO insert logic for calling script(s) here
+        }
+        else {
+          sh "ansible-playbook -i \'../${slave.inventory}\' ${TEST_DIR}/ansible-playbooks/*/playbook.yml"
```
Comment (scoheb) [inline Jenkinsfile]: should be \'/${slave.inventory}\'
that path is absolute.

File: Jenkinsfile [hunk 3]
```diff
@@ -1,91 +1,117 @@
...
-                      userRemoteConfigs:
-                              [[url: "${params.LIBRARIES_REPO}"]]])
+identifier: "multiarch-ci-libraries@${params.LIBRARIES_VERSION}",
+retriever: modernSCM([$class: 'GitSCMSource',remote: "${params.LIBRARIES_REPO}"])
 
 parallelMultiArchTest(
-        arches,
-        provisioningConfig.create(params),
+  arches,
+  provisioningConfig.create(),
```
Comment (scoheb) [inline Jenkinsfile]: the create method in dev branch needs params as a parameter

File: Jenkinsfile [hunk 4]
```diff
@@ -1,91 +1,117 @@
...
 
 library changelog: false,
-        identifier: "multiarch-ci-libraries@${params.LIBRARIES_BRANCH}",
-        retriever: legacySCM([$class: 'GitSCM', branches: [[name: "*/${params.LIBRARIES_BRANCH}"]],
-                      doGenerateSubmoduleConfigurations: false,
-                      extensions: [], submoduleCfg: [],
-                      userRemoteConfigs:
-                              [[url: "${params.LIBRARIES_REPO}"]]])
+identifier: "multiarch-ci-libraries@${params.LIBRARIES_VERSION}",
+retriever: modernSCM([$class: 'GitSCMSource',remote: "${params.LIBRARIES_REPO}"])
```
Comment (scoheb) [inline Jenkinsfile]: curious to see if this works when using a branch or tag.

**Review by scoheb — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 2 above)
Comment (jaypoulz) [reply Jenkinsfile]: does it need the leading slash if we're expected an absolute path slave inventory?
Shouldn't it be just be `'${slave.inventory}' ` ?

File: Jenkinsfile (hunk 1 above)
Comment (jaypoulz) [reply Jenkinsfile]: good question - I will test this 👍

File: Jenkinsfile (hunk 2 above)
Comment (scoheb) [reply Jenkinsfile]: yes. sorry typo :)

File: Jenkinsfile (hunk 4 above)
Comment (jaypoulz) [reply Jenkinsfile]: I will test this 👍

Comment (jaypoulz) [reply Jenkinsfile]: Testing shows that this works for both branches and tags. It works similiar to how OpenShift git ref works when specify source for a template. I think I will change the name of the variables like this to reference to remain consistent with OpenShift.

**Review by scoheb — APPROVED** [review]

**Pull request merged**